SeaPublicWebsite/Controllers/EnergyEfficiencyController.cs

```diff
@@ -1126,8 +1136,27 @@ public async Task<IActionResult> AnswerSummary_PostAsync(string reference)
         }
 
         [HttpGet("returning-user/{reference}")]
-        public async Task<IActionResult> ReturningUser_Get(string reference)
+        public async Task<IActionResult> ReturningUser_Get(string reference, bool fromMagicLink = true)
         {
+            if (cookieService.HasAcceptedGoogleAnalytics(Request))
+            {
+                await googleAnalyticsService.SendEvent(new GaRequestBody
+                {
+                    ClientId = Guid.NewGuid().ToString(),

```

Comment:
I might be able to dig up the client ID if I look into the cookies, but it's not needed

Reply:
What is the client ID used for?

Reply:
So that we know which user caused the event. I will dig up the id from the _ga cookie and use it here instead